src/main/java/edu/ucsb/cs56/ucsb_courses_search/service/UCSBQuarterCalendarService.java

```diff
@@ -0,0 +1,63 @@
+package edu.ucsb.cs56.ucsb_courses_search.service;
+
+import java.util.Arrays;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Service object that wraps the UCSB Academic Curriculum API
+ */
+@Service
+public class UCSBQuarterCalendarService implements CalendarService {
+
+    private Logger logger = LoggerFactory.getLogger(UCSBQuarterCalendarService.class);
+
+    private String apiKey;
+
+    public UCSBQuarterCalendarService(@Value("${ucsb.api.consumer_key}") String apiKey) {
+        this.apiKey = apiKey;
+        logger.info("apiKey=" + apiKey);
```

Comment:
This log statement absolutely needs to be removed. This publicly leaks the API key. You need to reset your UCSB API key. Never log any secret key. I know this was done elsewhere in the code, but it's a huge bug and security risk.